contracts/src/SP1ICS07Tendermint.sol

```diff
@@ -331,52 +349,84 @@ contract SP1ICS07Tendermint is
         if (clientState.isFrozen) {
             revert FrozenClientState();
         }
-        if (output.env.now > block.timestamp) {
-            revert ProofIsInTheFuture(block.timestamp, output.env.now);
+        validateEnv(output.env);
+
+        bytes32 outputConsensusStateHash = keccak256(abi.encode(output.trustedConsensusState));
+        bytes32 trustedConsensusStateHash = getConsensusStateHash(output.trustedHeight.revisionHeight);
+        if (outputConsensusStateHash != trustedConsensusStateHash) {
+            revert ConsensusStateHashMismatch(trustedConsensusStateHash, outputConsensusStateHash);
+        }
+    }
+
+    /// @notice Validates the SP1ICS07MisbehaviourOutput public values.
+    /// @param output The public values.
+    function validateMisbehaviourOutput(MisbehaviourOutput memory output) private view {
+        if (clientState.isFrozen) {
+            revert FrozenClientState();
+        }
+        validateEnv(output.env);
+
+        bytes32 outputConsensusStateHash1 = keccak256(abi.encode(output.trustedConsensusState1));
```

Comment:
Here we basically verify that the trusted consensus state in both headers are known (trusted) by us already.

Reply:
In code docs will make it clearer for new readers why this is needed